src/kakarot/library.cairo

```diff
@@ -295,12 +292,7 @@ namespace Kakarot {
             bytecode_len=return_data_len,
             bytecode=return_data,
         );
-        // Increment nonce
-        IContractAccount.increment_nonce(contract_address=starknet_contract_address);
```

Comment:
I think might be what's causing the ef test to fail

Reply:
Hmmm, true, this PR shouldn't touch this part of the file, might be something went wrong during rebase!

Reply:
you should keep this indeed, good we have the test!